token/program2/src/processor.rs

```diff
@@ -149,6 +157,18 @@ impl Processor {
             return Err(TokenError::MintMismatch.into());
         }
 
+        if let Some((mint_account_info, decimals)) = mint_account_info {
```

Comment:
This could be a u8 check between the passed valuea nd the src account since we don't really need to validate mint, we just want to express the decimals in the instruction data

Reply:
Why we need `mint` in `Transfer2`:
When signing a `Transfer2` on a ledger device, the ledger program has no access to the on-chain accounts.   The ledger program may only choose to support well-known mints (at which point it knows `decimals` and technically doesn't need to include them in `Transfer2`).   But it definitely wants to assert the `mint` of the the to/from token accounts.

Why we need `decimals` in  `Transfer2`:
You try to generically sign a token transfer based on mint data received over RPC.   But you can't trust that RPC isn't lying to you and turning that 0.1 token into a 100000 token.   So in this case, you need to assert both the mint and decimals.

Reply:
Yeah, makes sense, writing a client helps expose these use cases